connection.go

```diff
@@ -25,74 +25,30 @@ type Conn interface {
 	RemoteHeader() http.Header
 }
 
-type writePacket struct {
-	header parser.Header
-
-	data []interface{}
-}
-
-type namespaces struct {
-	namespaces map[string]*namespaceConn
-	mu         sync.RWMutex
-}
-
-func newNamespaces() *namespaces {
-	return &namespaces{
-		namespaces: make(map[string]*namespaceConn),
-	}
-}
-
-func (n *namespaces) Get(ns string) (*namespaceConn, bool) {
-	n.mu.RLock()
-	defer n.mu.RUnlock()
-	namespace, ok := n.namespaces[ns]
-	return namespace, ok
-}
-
-func (n *namespaces) Set(ns string, conn *namespaceConn) {
-	n.mu.Lock()
-	defer n.mu.Unlock()
-	n.namespaces[ns] = conn
-}
-
-func (n *namespaces) Delete(ns string) {
-	n.mu.Lock()
-	defer n.mu.Unlock()
-	delete(n.namespaces, ns)
-}
-
-func (n *namespaces) Range(fn func(ns string, nc *namespaceConn)) {
-	n.mu.RLock()
-	defer n.mu.RUnlock()
-	for ns, nc := range n.namespaces {
-		fn(ns, nc)
-	}
-}
-
 type conn struct {
+	*parser.Encoder
+	*parser.Decoder
```

Comment:
It exposes Encode/Decode to users. 

I know it's convenient to write `conn.Encode()` than `conn.encoder.Encode()`, but I hope exposing methods could give a guide that which one could be called by caller, and which one should not. Exposing Encode/Decode breaks that rule.

Reply:
> It exposes Encode/Decode to users.

Sure, sorry, my fault. Rollbacked